Configure error when Fortran is not enable

Using `-DTrilinos_ENABLE_Fotran=OFF` should disable the fortran api. Right now it produces the following error:
CMake Error: CMake can not determine linker language for target: dtk_fortran_api
CMake Error in DataTransferKit/packages/Adapters/Fortran_API/src/CMakeLists.txt:
  Exporting the target "dtk_fortran_api" is not allowed since its linker
  language cannot be determined

Here is how to reproduce what @Rombur reported.
```
[container]$ cd $TRILINOS_DIR/DataTransferKit
[container]$ mkdir build && cd build
[container]$ ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
```
which yield
```
[...]
-- Configuring done
CMake Error: CMake can not determine linker language for target: dtk_fortran_api
CMake Error in DataTransferKit/packages/Adapters/Fortran_API/src/CMakeLists.txt:
  Exporting the target "dtk_fortran_api" is not allowed since its linker
  language cannot be determined


CMake Error: CMake can not determine linker language for target: dtk_fortran_api
CMake Error in DataTransferKit/packages/Adapters/Fortran_API/src/CMakeLists.txt:
  Exporting the target "dtk_fortran_api" is not allowed since its linker
  language cannot be determined


CMake Error: Cannot determine link language for target "dtk_fortran_api".
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_Fortran_LINK_EXECUTABLE
-- Generating done
-- Build files have been written to: /scratch/source/trilinos/release/DataTransferKit/build
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,6 +72,17 @@ else()
 endif()
 
 ##---------------------------------------------------------------------------##
+## Disable Fortran_API if Trilinos_ENABLE_Fortran is FALSE
+##---------------------------------------------------------------------------##
+IF (NOT ${PROJECT_NAME}_ENABLE_Fortran)
+    IF (${${PROJECT_NAME}_ENABLE_${PACKAGE_NAME}Fortran_API})
+        MESSAGE("-- " "NOTE: Setting ${PROJECT_NAME}_ENABLE_${PACKAGE_NAME}Fortran_API=OFF because"
+                "${PROJECT_NAME}_ENABLE_Fortran = '${${PROJECT_NAME}_ENABLE_Fortran}'")
+        SET(${PROJECT_NAME}_ENABLE_${PACKAGE_NAME}Fortran_API OFF)
+    ENDIF()
+ENDIF()
+
+##---------------------------------------------------------------------------##
 ## Do standard postprocessing
 ##---------------------------------------------------------------------------##
 

